### PR TITLE
Determine pre-upgrade image automatically

### PR DESCRIPTION
**Pull request by jgrassler**
The oncontroller_testpreupgrade() function had hardwired
glance image names for Cloud 8 and Cloud 9 that might not
work, depending on which image URL the Tempest barclamp uses
to create its test images. In that case the function would
be unable to create the Heat stack for upgrade testing and
fail the upgrade on the Cloud 8 side. This commit replaces the
hardwired image name by a heuristic that picks the correct
image name based on the images that are actually present.

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -5047,11 +5047,17 @@ function ping_fips
 # Use heat_stack_params to provide parameters to heat template
 function oncontroller_testpreupgrade
 {
-    local image=cirros-0.3.4-x86_64-tempest-machine
-    iscloudver 8plus && image=cirros-0.4.0-x86_64-tempest-machine
+    local tempest_image=$(openstack image list -f value | cut -d ' ' -f 2 | grep '^cirros-.*-tempest$' | head -n 1)
```
Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: Will `openstack image list` work for the oldest upgrade scenario we support?

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -5047,11 +5047,17 @@ function ping_fips
...
+    if [ -z "$tempest_image" ]; then
+        # Depending on URL in the tempest barclamp, this may be, the image's name so retry if we can't find the above:
```
Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: I do not understand the sentence...

File: scripts/qa_crowbarsetup.sh (hunk 1 above)
Comment (jsuchome) [reply scripts/qa_crowbarsetup.sh]: Update: yes, it works for SOC7, which I believe should cover the oldest case

File: scripts/qa_crowbarsetup.sh (hunk 2 above)
Comment (jgrassler) [reply scripts/qa_crowbarsetup.sh]: Indeed...me neither...

Better now?

**Pull request merged**